napari/layers/base/base.py

```diff
@@ -634,6 +634,53 @@ def _dims_displayed_order(self):
         order = sorted(range(len(displayed)), key=lambda x: displayed[x])
         return tuple(order)
 
+    @property
+    def _data_dims_order(self):
+        """Order of the axes into the raw data array.
+
+        For affines without permutation, it will be the same as _dims_order.
+        """
+        if self._data_to_world.is_permutation:
+            perm = self._data_to_world.perm
+            return [perm[d] for d in self._dims_order]
+        return self._dims_order
```

Comment:
I know some linters remove `else`s in situations like this, but I find it easier to read. In my first reading I'd missed the `return` inside the `if` block and it seemed like you were ignoring the work in that block.

```suggestion
        if self._data_to_world.is_permutation:
            perm = self._data_to_world.perm
            return [perm[d] for d in self._dims_order]
        else:
            return self._dims_order
```

Alternatively:

```python
        if self._data_to_world.is_permutation:
            perm = self._data_to_world.perm
            dims_order = [perm[d] for d in self._dims_order]
        else:
            dims_order = self._dims_order
        return dims_order
```